src/coprocessor/codec/mysql/decimal.rs

```diff
@@ -120,7 +120,15 @@ macro_rules! word_cnt {
         word_cnt!($len, u8)
     };
     ($len:expr, $t:ty) => {{
-        (($len) + DIGITS_PER_WORD as $t - 1) / DIGITS_PER_WORD as $t
+        if $len > 0 && $len as usize > (DIGITS_PER_WORD * WORD_BUF_LEN) as usize {
+            // process overflow
+            (WORD_BUF_LEN + 1) as $t
+        } else if $len <= 0 && ($len as $t) > 0 {
```

Comment:
In the past, the type of `$len` is `$t`.

Reply:
yes. But bug exists when we try to convert $len from signed to unsigned before this function, so we'd like to process this case here.